### DISCUSSION TITLE
I2C Communication between two ESP/Pico possible?

### DISCUSSION THREAD

**Discussion by ingelumer** [micropython/micropython#11278] [General]
Hello all,

I am searching a way to connect two MicroPython devices (ESP32/Pico) by wire and without a WiFi connection.

My target is to send water level data from a SR04T sensor in my Water tank from one device to the other every 15 minutes.
Since in the tank there is no WiFi signal but a 3 pair cable (~4m), I thought it could be an idea if data could be sent from the tank to another device in my garage.

Is it possible to do this by using connected I2C PINs ?

Thanks for any ideas :-)

**Comment by robert-hh** [comment]
Use UART. I2C is only suitable for short links, and there is not peripheral I2C mode implemented.
